filemgr/src/main/java/org/apache/oodt/cas/filemgr/catalog/LuceneCatalog.java

```diff
@@ -819,15 +833,21 @@ public Metadata getReducedMetadata(Product product, List<String> elements) throw
      * 
      * @see org.apache.oodt.cas.filemgr.catalog.Catalog#getValidationLayer()
      */
-    public ValidationLayer getValidationLayer() {
-        return valLayer;
+    @Override
+    public ValidationLayer getValidationLayer() throws CatalogException {
+        if(valLayer != null) {
+            return valLayer;
+        } else {
+            throw new CatalogException("validation layer is not initialized");
```

Comment:
Let's keep this method unchanged since it is better we modify the interface `Catalog`. Instead let's add null checks when this method is being used.